lib/classes/map_coordinates.rb

```diff
@@ -0,0 +1,44 @@
+class MapCoordinates
+  def initialize(location)
+    @location = location.gsub(" ", "%20")
+  end
+
+  def coordinates
+    response = JSON.parse(HTTP.get(api_url).as_json["body"][0])
+    city_country = formatted_city_country(response["candidates"][0]["formatted_address"])
+    location = city_country.merge(formatted_lat_lon(response["candidates"][0]["geometry"]["location"]))
+  end
```

Comment:
something i like to do in situations like this is break out new methods instead of setting variabels:
```ruby
def response
  @response ||= JSON.parse(HTTP.get(api_url).as_json["body"][0])
end
```
Having the response cached and available in a method also comes in handy if you wanted to play around with the response in a rails console or pry instance.